### ISSUE TITLE
New window warning aria-label incorrect for links with images and text in link

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1268]
If a link has both a decorative image and also text in in the aria-label that is being created by the new window warnings fix fails to get the link text, resulting in inaccessible links that fail WCAG Name in Label.

If there is an image and text in a link, the `aria-label` for the link needs to use both.

## Failure 1

Example code in editor:
`<p><a href="https://www.w3.org/TR/WCAG22/" target="_blank" rel="noreferrer noopener"><img class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt=""><br>Visit the WCAG 2.2 website </a></p>`

This results in a link like this:
`<a href="https://www.w3.org/TR/WCAG22/" target="_blank" rel="noreferrer noopener" aria-label="opens a new window" data-nww-processed="true"><img decoding="async" width="203" height="173" class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt=""><br>Visit the WCAG 2.2 website <i class="edac-nww-external-link-icon" aria-hidden="true"></i></a>`

The expected result should be:
`<a href="https://www.w3.org/TR/WCAG22/" target="_blank" rel="noreferrer noopener" aria-label="Visit the WCAG 2.2 website, opens a new window" data-nww-processed="true"><img decoding="async" width="203" height="173" class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt=""><br>Visit the WCAG 2.2 website <i class="edac-nww-external-link-icon" aria-hidden="true"></i></a>`

## Failure 2
Example code in editor:
`<p><a href="https://www.w3.org/TR/WCAG22/" target="_blank" rel="noreferrer noopener"><img class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt="meaningful image"><br>Visit the WCAG 2.2 website </a></p>`

Resulting link code:
`<a href="https://www.w3.org/TR/WCAG22/" target="_blank" rel="noreferrer noopener" aria-label="meaningful image, opens a new window" data-nww-processed="true"><img decoding="async" width="203" height="173" class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt="meaningful image"><br>Visit the WCAG 2.2 website <i class="edac-nww-external-link-icon" aria-hidden="true"></i></a>`

Expected link code:
`<a href="https://www.w3.org/TR/WCAG22/" target="_blank" rel="noreferrer noopener" aria-label="meaningful image Visit the WCAG 2.2 website, opens a new window" data-nww-processed="true"><img decoding="async" width="203" height="173" class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt="meaningful image"><br>Visit the WCAG 2.2 website <i class="edac-nww-external-link-icon" aria-hidden="true"></i></a>`

This is high priority as it's breaking front-end accessibility in some cases.

**Comment by pattonwebz** [comment]
@amberhinds if you can [test this on playground](https://playground.wordpress.net/?blueprint-url=data%3Aapplication%2Fjson%3Bbase64%2CeyJsYW5kaW5nUGFnZSI6Ii93cC1hZG1pbi9wbHVnaW5zLnBocCIsInN0ZXBzIjpbeyJzdGVwIjoibG9naW4ifSx7InN0ZXAiOiJpbnN0YWxsUGx1Z2luIiwicGx1Z2luWmlwRmlsZSI6eyJyZXNvdXJjZSI6InVybCIsInVybCI6Imh0dHBzOi8vbmlnaHRseS5saW5rL2VxdWFsaXplZGlnaXRhbC9hY2Nlc3NpYmlsaXR5LWNoZWNrZXIvYWN0aW9ucy9ydW5zLzI3Nzk4OTExODQzL2FjY2Vzc2liaWxpdHktY2hlY2tlci0xLjQzLjAtMTc3My03MGMwYTU4Ny56aXAifX0seyJzdGVwIjoiYWN0aXZhdGVQbHVnaW4iLCJwbHVnaW5QYXRoIjoiYWNjZXNzaWJpbGl0eS1jaGVja2VyL2FjY2Vzc2liaWxpdHktY2hlY2tlci5waHAifV19) for me that would be great. I tested with your markup and got the results below. It seems like it is the correct label you would expect with the fix after the changes.

```
<p>
    <a 
      href="https://www.w3.org/TR/WCAG22/" 
      target="_blank" 
      rel="noreferrer noopener" 
      aria-label="Visit the WCAG 2.2 website, opens a new window" 
      data-nww-processed="true"
    >
        <img decoding="async" class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt="">
        <br>Visit the WCAG 2.2 website 
        <i class="edac-nww-external-link-icon" aria-hidden="true"></i>
    </a>
</p>

<p>
    <a
      href="https://www.w3.org/TR/WCAG22/" 
      target="_blank" 
      rel="noreferrer noopener" 
      aria-label="meaningful image Visit the WCAG 2.2 website, opens a new window"
      data-nww-processed="true"
    >
        <img decoding="async" class="wp-image-3330" style="width: 150px;box-shadow:none !important;" src="https://equalizedigital.com/wp-content/uploads/2020/04/user-testing-icon.png" alt="meaningful image">
        <br>Visit the WCAG 2.2 website 
        <i class="edac-nww-external-link-icon" aria-hidden="true"></i>
    </a>
</p>
```

**Comment by amberhinds** [comment]
@pattonwebz tested and confirmed this solves the issue. Thank you so much!!
